### PR TITLE
Update the URL.path for the Reverse Proxy with the incoming URL.path

### PR DESCRIPTION
**Pull request by spaparaju**
Update the URL.path for the Reverse Proxy with the incoming URL.path

### CODE REVIEW

**Comment by squat** [comment]
I don't think this is exactly the semantics we want. We likely want something like what NewSingleHostReverseProxy does, where the path is a combination of the configured URL's path and the request's URL's path: https://golang.org/src/net/http/httputil/reverseproxy.go?s=4447:4508#L114

**Comment by spaparaju** [comment]
Ack! It's nasty detour (to fix Host related bug) I was trying to avoid duplicating the entire functionality of `NewSingleHostReverseProxy()`.

File: main.go [hunk 1]
```diff
@@ -209,11 +210,44 @@ func main() {
...
+			// Create Reverse Proxy.
 			p := httputil.ReverseProxy{
 				Director: func(request *http.Request) {
 					request.URL.Scheme = cfg.url.Scheme
+					// Set the Host at both request and request.URL objects.
 					request.Host = cfg.url.Host
 					request.URL.Host = cfg.url.Host
+					// Derive path from the paths of configured URL and request URL.
+					request.URL.Path, request.URL.RawPath =
+						func(a, b *url.URL) (path, rawpath string) {
```
Comment (squat) [inline main.go]: can we extract this anonymous function into its own func? I think this would help with readability

**Comment by squat** [comment]
looks fine overall! plz extract the func and then let's merge it :rocket:

**Review by squat — APPROVED** [review]
Lgtm!

**Pull request merged**